src/wizards/deployToOpenShift/deployToOpenShift.ts

```diff
@@ -0,0 +1,61 @@
+import { commands, Uri, window } from "vscode";
+import { ProjectLabelInfo } from "../../definitions/ProjectLabelInfo";
+import { getOpenShiftConnector, installOpenShiftConnectorWithProgress, isOpenShiftConnectorInstalled, OPENSHIFT_CONNECTOR } from "../../utils/openShiftConnectorUtils";
+import { getQuarkusProject } from "../getQuarkusProject";
+
+/**
+ * Attempts to select a Quarkus project, then deploy it to OpenShift using OpenShift Connector
+ */
+export async function deployToOpenShift() {
+  try {
+    await installOpenShiftConnectorIfNeeded();
+    const quarkusProject: ProjectLabelInfo = await getQuarkusProject();
+    const result: string = await commands.executeCommand("openshift.component.createFromRootWorkspaceFolder", //
```

Comment:
I am looking into trying to push the component, then activating the "create component" if that fails. I am hoping this will cover the use case where there is a `devfile.yaml` and a `.odo` folder, but the component is missing from OpenShift. I'm also going to wrap the command calls to a different interface to make this function cleaner to read.

Reply:
I changed strategies to checking the folder for `.odo` to detect if the component should be created or just pushed.